### PR TITLE
Add form validation in Dataman App

### PR DESCRIPTION
**Pull request by sameera222**
This Pull Request will: 
- [x]  Add form validation to the dataman app

### CODE REVIEW

File: apps/Dataman/src/Utils/PurchaseValidations.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            </View>
+            <View className="w-2/5 pl-2">
+              <Text>Unit</Text>
+              <Controller
+                control={control}
+                rules={{
+                  required: "This field is required",
+                  pattern: {
+                    value:
+                      /^(1\s|1|)?((\(\d{3}\))|\d{3})(\\-|\s)?(\d{3})(\\-|\s)?(\d{4})$/,
```
Comment (sutyum) [inline apps/Dataman/src/Utils/PurchaseValidations.tsx]: refactor these out and give them reasonable names such as phoneRegex

File: apps/Dataman/src/Utils/StoreValidations.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            <View className="flex-[1]">
+              <Text className="text-base">Request Quantity</Text>
+
+              <Controller
+                control={control}
+                rules={{
+                  required: "This field is required",
+                  pattern: {
+                    value:
+                      /^(1\s|1|)?((\(\d{3}\))|\d{3})(\\-|\s)?(\d{3})(\\-|\s)?(\d{4})$/,
```
Comment (sutyum) [inline apps/Dataman/src/Utils/StoreValidations.tsx]: create a seperate file for regexes

**Review by sutyum — CHANGES_REQUESTED** [review]

File: apps/Dataman/src/Utils/PurchaseValidations.tsx (hunk 1 above)
Comment (sameera222) [reply apps/Dataman/src/Utils/PurchaseValidations.tsx]: This code has been refactored

File: apps/Dataman/src/Utils/StoreValidations.tsx (hunk 2 above)
Comment (sameera222) [reply apps/Dataman/src/Utils/StoreValidations.tsx]: done.

**Comment by sutyum** [comment]
https://www.loom.com/share/cae2001ed3344ef88e4c6b5d9eb38dae

**Review by sutyum — APPROVED** [review]

**Pull request merged**